jdaviz/core/template_mixin.py

```diff
@@ -4829,6 +4829,48 @@ def clear_table(self):
         self._qtable = None
         self._plugin.session.hub.broadcast(PluginTableModifiedMessage(sender=self))
 
+    def select_rows(self, rows):
+        """
+        Select rows from the current table by index, indices, or slice.
+
+        This method first clears the currently selected rows by resetting 
+        `self.selected_rows` to an empty list. The rows specified by the input
+        `rows` are then applied as the new selection.
+
+        Parameters
+        ----------
+        rows : int, list of int, slice, or tuple of slice
+            The rows to select. This can be:
+            - An integer specifying a single row index.
+            - A list of integers specifying multiple row indices.
+            - A slice object specifying a range of rows.
+            - A tuple of slices (e.g using np.s_)
+
+        """
+
+        if isinstance(rows, tuple):
+            # if 'rows' are a numpy slice, then it could be
+            # a tuple of slices.
+            if isinstance(rows[0], slice):
+                selected = []
+                for sl in rows:
+                    selected.append(self.items[rows][0])
+            else:
+                selected = self.items[rows]
+
+
+        elif isinstance(rows, (slice, list)):
+            selected = self.items[rows]
+        elif isinstance(rows, int):
+            selected = [self.items[rows]]
+
+        # first, deselect to revert current selection
+        self.selected_rows = []
+
+        # then apply current new selection
+        self.selected_rows = selected
```

Comment:
will this work (or is there another reason it needs to be emptied first)?

```suggestion
        # then apply current new selection
        self.selected_rows = selected
        self.send_state('selected_rows')
```